src/tck/Reactive.Streams.TCK.Tests/RangePublisherTest.cs

```diff
@@ -0,0 +1,192 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Reactive.Streams.TCK.Tests
+{
+    [TestFixture]
+    public class RangePublisherTest : PublisherVerification<int>
+    {
+        static readonly ConcurrentDictionary<int, string> stacks = new ConcurrentDictionary<int, string>();
+
+        static readonly ConcurrentDictionary<int, bool> states = new ConcurrentDictionary<int, bool>();
+
+        static int id;
+
+        [TearDown]
+        public void AfterTest()
+        {
+            bool fail = false;
+            StringBuilder b = new StringBuilder();
+            foreach (var t in states)
+            {
+                if (!t.Value)
+                {
+                    b.Append("\r\n-------------------------------");
+
+                    b.Append("\r\nat ").Append(stacks[t.Key]);
```

Comment:
It printed the methods that needed attention when I was fixing the underlying problem. Plus it will print problems in case new TCK tests are added in the future.